File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Pair {
    int temperature;
    int index;
    Pair(int temperature, int index) {
        this.temperature = temperature;
        this.index = index;
    }
}
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        Stack<Pair> stack = new Stack<>();
        int[] res = new int[n];
        stack.push(new Pair(temperatures[0], 0));

        for(int i = 1; i < n; i++) {
            while(!stack.isEmpty() && temperatures[i] > stack.peek().temperature) {
                res[stack.peek().index] = i - stack.peek().index;
                stack.pop();
            }
            stack.push(new Pair(temperatures[i], i));
        }

        return res;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        ds = []

        def helper(index, target):
            if index >= len(nums):
                if target == 0:
                    res.append(ds.copy())
                return
            
            # not-pick
            helper(index+1, target)

            # pick
            if (target - nums[index]) >= 0:
                ds.append(nums[index])
                helper(index, target - nums[index])
                ds.pop()
        
        helper(0, target)
        return res
